### PR TITLE
docs(array filter): add example using array filters

### PR DESCRIPTION
**Pull request by jlord**
Add an example of using `arrayFilters` in an update for the docs team.

Fixes [NODE-1263](https://jira.mongodb.org/browse/NODE-1263)

### CODE REVIEW

**Comment by mbroadst** [comment]
These should be added to the `operation_example_tests.js`, `operation_generators_example_tests.js`, and `operation_promises_example_tests.js` files instead (modified for each variant). It was okay to add a file just for change streams because there were a number of tests, but this is just a single test.

File: test/functional/operation_array_update_example_test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    },
+
+    test: function(done) {
+      const configuration = this.configuration;
+      const client = configuration.newClient(configuration.writeConcernMax(), { poolSize: 1 });
+      client.connect(function(err, client) {
+        const db = client.db(configuration.db);
+        const collection = db.collection('arrayFilterUpdateExample');
+
+        // 3. Exploiting the power of arrays
```
Comment (mbroadst) [inline test/functional/operation_array_update_example_test.js]: also does this line need a corresponding `end` tag?

Comment (jlord) [reply test/functional/operation_array_update_example_test.js]: I will check with the docs team 👍

**Comment by jlord** [comment]
I have moved the example to `examples_tests.js` which we talked about (in real life) of being a better home than the files mentioned ☝️ above. 

- [ ] I'm still waiting for feedback on if there is a closing code comment we need to use.

File: test/functional/examples_tests.js [hunk 2]
```diff
@@ -1192,4 +1192,36 @@ describe('Examples', function() {
       });
     }
   });
+
+  it.only('supports array filters when updating', {
```
Comment (daprahamian) [inline test/functional/examples_tests.js]: `.only`

Comment (jlord) [reply test/functional/examples_tests.js]: 🤣

**Pull request merged**